Dockerfile-alpine.template

```
@@ -190,30 +181,43 @@ RUN set -eux; \
 # Compile & install Erlang/OTP
 	make -j "$(getconf _NPROCESSORS_ONLN)" GEN_OPT_FLGS="-O2 -fno-strict-aliasing"; \
 	make install; \
-	cd ..; \
-	rm -rf \
-		"$OTP_PATH"* \
-		/usr/local/lib/erlang/lib/*/examples \
-		/usr/local/lib/erlang/lib/*/src \
-	; \
 	\
-	runDeps="$( \
+# Remove unnecessary files
+	find /usr/local/lib/erlang -type d -name examples -exec rm -rf '{}' +; \
+	find /usr/local/lib/erlang -type d -name src -exec rm -rf '{}' +; \
+	find /usr/local/lib/erlang -type d -name include -exec rm -rf '{}' +; \
+	\
+# Check that Erlang/OTP crypto & ssl were compiled against OpenSSL correctly
+	erl -noshell -eval 'io:format("~p~n~n~p~n~n", [crypto:supports(), ssl:versions()]), init:stop().'
+
+FROM alpine:3.16
+
+COPY --from=erlang-builder /usr/local/bin/ /usr/local/bin/
+COPY --from=erlang-builder /usr/local/etc/ssl/ /usr/local/etc/ssl/
+COPY --from=erlang-builder /usr/local/lib/ /usr/local/lib/
+
+ENV RABBITMQ_DATA_DIR=/var/lib/rabbitmq
+
+RUN set -eux; \
+# Configure OpenSSL to use system certs
+	ln -vsf /etc/ssl/certs /etc/ssl/private /usr/local/etc/ssl; \
+	\
+# Install run-time dependencies
+	readonly runDeps="$( \
 		scanelf --needed --nobanner --format '%n#p' --recursive /usr/local \
 			| tr ',' '\n' \
 			| sort -u \
-			| awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
+			| awk 'system("test -e /usr/local/lib/" $1) == 0 { next } { print "so:" $1 }' \
 	)"; \
```

Doing `readonly` this way won't interact with `set -e` correctly, so it needs to be after the variable is set:

```console
$ docker run --rm alpine:3.16 sh -c 'set -eux; readonly foo="$(false)"; echo oops'
+ false
+ readonly 'foo='
+ echo oops
oops

$ docker run --rm alpine:3.16 sh -c 'set -eux; foo="$(false)"; readonly foo; echo oops'
+ false
+ foo=

$ docker run --rm alpine:3.16 sh -c 'set -eux; foo="$(echo nice)"; readonly foo; echo "$foo"'
+ echo nice
+ foo=nice
+ readonly foo
+ echo nice
nice
```

```suggestion
	runDeps="$( \
￼		scanelf --needed --nobanner --format '%n#p' --recursive /usr/local \
￼			| tr ',' '\n' \
￼			| sort -u \
￼			| awk 'system("test -e /usr/local/lib/" $1) == 0 { next } { print "so:" $1 }' \
￼	)"; \
	readonly runDeps; \
```

---

Good catch. No reason to use `readonly` tbh